enhancements/storage/csi-vsphere-operator.md

```diff
@@ -105,33 +105,49 @@ Currently while deploying Openshift a user can configure datastore used by OCP v
 
 To solve this problem vsphere CSI operator is going to create a storagePolicy by tagging selected datastore in the installer. This will require OCP to have expanded permissions of creating storagePolicies. After creating the storagePolicy, the vSphere CSI operator will also create corresponding storageclass.
 
-If CSI operator can not create storage Policy for some reason:
+###### Error handling during creation of storage policy and storage class
 
-- The operator will mark itself as disabled and stop further driver installation.
-- It will periodically retry installation and wait for admin to grant permissions to create StoragePolicy.
-- Create a metric if storagePolicy and storageClass creation fails in 4.7.
+If CSI operator can not create storage Policy or storageClass for some reason:
 
-##### Hardware and vCenter version handling
+Cluster will be degraded when:
+- We can't talk to Kube API server and for some reason can't read secret or can't create storageclass etc.
 
-When vSphere CSI operator starts, using credentials provided by cluster-storage-operator, it will first verifiy vCenter version and HW version of VMs. If vCenter version is not 6.7u3 or greater and HW version is not 15 or greater on all VMs and this is a fresh install(i.e there is no `CSIDriver` object already installed by this operator) - it will:
+Cluster will be marked as un-upgradable when:
+- There are any vCenter related errors (connection refused, 503 errors or permission denied)
 
-1. Mark itself as disabled, stop CSI driver installation.
-2. Add clear message to indicate the error.
-3. It will keep retrying installation with exponential backoff assuming admin manually fixes the HW version of VMs.
+In case operator is marked as un-upgradeable for some reason - detailed information will be added to `ClusterCSIDriver` object and an appropriate metric will be emitted.
```

Comment:
We should mention that the operator (and whole cluster) will be still `Available=true` to proceed with installation / cluster upgrade.